src/components/dialog/Dialog.jsx

```diff
@@ -69,6 +76,7 @@ function BaseDialog({
   const overlayRef = React.useRef(null);
   const containerRef = React.useRef(null);
   const [exiting, setExiting] = React.useState<boolean>(false);
+  const [id] = React.useState(`dialog-${getRandomId()}`);
```

Comment:
ℹ️ Instead of using regular `const` here I'm using state to avoid regenerating id here if component state changes. This wouldn't affect how proposed solution works, but I think the expected behaviour here would be to have one constant id until component is unmounted

Reply:
useRef should be used here since we don't need reset the value during component lifetime.